src/plugins/workspace/public/components/workspace_list/index.tsx

```diff
@@ -38,23 +45,100 @@ export interface WorkspaceListProps {
   registeredUseCases$: BehaviorSubject<WorkspaceUseCase[]>;
 }
 
+interface WorkspaceAttributeWithUseCaseID extends WorkspaceAttribute {
+  useCase?: string;
+}
+
 export const WorkspaceList = ({ registeredUseCases$ }: WorkspaceListProps) => {
   const {
     services: { workspaces, application, http },
   } = useOpenSearchDashboards();
   const registeredUseCases = useObservable(registeredUseCases$);
   const isDashboardAdmin = application?.capabilities?.dashboards?.isDashboardAdmin;
-
   const initialSortField = 'name';
   const initialSortDirection = 'asc';
   const workspaceList = useObservable(workspaces?.workspaceList$ ?? of([]), []);
-  const [queryInput, setQueryInput] = useState<string>('');
+
   const [pagination, setPagination] = useState({
     pageIndex: 0,
     pageSize: 5,
     pageSizeOptions: [5, 10, 20],
   });
-  const [deletedWorkspace, setDeletedWorkspace] = useState<WorkspaceAttribute | null>(null);
+  const [deletedWorkspaces, setDeletedWorkspaces] = useState<WorkspaceAttribute[]>([]);
+  const [selection, setSelection] = useState<WorkspaceAttribute[]>([]);
+
+  const extractUseCaseFromFeatures = (features: string[]) => {
+    if (!features || features.length === 0) {
+      return '';
+    }
+    const useCaseId = getFirstUseCaseOfFeatureConfigs(features);
+    const usecase =
+      useCaseId === DEFAULT_NAV_GROUPS.all.id
+        ? DEFAULT_NAV_GROUPS.all
+        : registeredUseCases?.find(({ id }) => id === useCaseId);
+    if (usecase) {
+      return usecase.title;
+    }
+  };
+
+  const newWorkspaceList: WorkspaceAttributeWithUseCaseID[] = useMemo(() => {
+    return workspaceList.map(
+      (workspace): WorkspaceAttributeWithUseCaseID => ({
+        ...workspace,
+        useCase: extractUseCaseFromFeatures(workspace.features ?? []),
+      })
+    );
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Why we use eslint ignore here?

Reply:
Hi, I add it to prevent a warning about not including extractUseCaseFromFeatures in the dependency array, which is indirectly dependent on workspaceList.features, and it is already accounted for through workspaceList in the dependency array

Reply:
Using eslint ignore in component file would not look reasonable. What about adding extractUseCaseFromFeatures to dependency or wrapping extractUseCaseFromFeatures with `useCallback` and adding it to dependency.

Reply:
Sure, I will update it, Thank you~